ext_py/tests/test_affine_subspace.py

```
@@ -0,0 +1,181 @@
+    assert list(s.offset()) == [1, 0, 0]
+    # [1,1,1] = offset + [0,1,1] is contained; [0,0,0] is not.
+    assert s.contains(fp.FpVector.from_py(2, [1, 1, 1]))
+    assert not s.contains(fp.FpVector.from_py(2, [0, 0, 0]))
+
+
+@pytest.mark.xfail(reason="panics")
+def test_invalid_construction_raises():
+    linear = make_subspace(2, [[0, 1, 0]])
+
+    # Offset length must match the linear part's ambient dimension.
+    with pytest.raises(ValueError):
```

Ruff (0.15.20)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)